cwms-data-api/src/main/java/cwms/cda/data/dao/project/ProjectChildrenDao.java

```diff
@@ -0,0 +1,262 @@
+
+        Set<String> kinds = getMatchingKinds(kindRegex, allKinds);
+
+        return children(office, projLike, kinds);
+    }
+
+    private List<ProjectChildren> children(String office, String projLike, Set<String> kinds) {
+
+        Map<String, ProjectChildren.Builder> builderMap = new LinkedHashMap<>();  // proj-id->
+
+        for (String kind : kinds) {
+            Map<String, List<CwmsId>> locsOfKind = getChildrenOfKind(office, projLike, kind);
+            if (locsOfKind != null) {
+                for (Map.Entry<String, List<CwmsId>> entry : locsOfKind.entrySet()) {
+                    String projId = entry.getKey();
+                    List<CwmsId> locs = entry.getValue();
+                    ProjectChildren.Builder builder = builderMap.computeIfAbsent(projId, k ->
+                            new ProjectChildren.Builder()
+                                    .withProject(new CwmsId.Builder()
+                                            .withOfficeId(office)
+                                            .withName(projId)
+                                            .build()));
+                    switch (kind) {
+                        case "EMBANKMENT":
+                            builder.withEmbankments(locs);
+                            break;
+                        case "LOCK":
+                            builder.withLocks(locs);
+                            break;
+                        case "OUTLET":
+                            builder.withOutlets(locs);
+                            break;
+                        case "TURBINE":
+                            builder.withTurbines(locs);
+                            break;
+                        case "GATE":
+                            builder.withGates(locs);
+                            break;
+                        default:
+                            break;
+                    }
+
+                }
+            }
+        }
+
+        return builderMap.values().stream()
+                .map(ProjectChildren.Builder::build)
+                .collect(Collectors.toList());
+
+    }
+
+    @Nullable
+    private Map<String, List<CwmsId>> getChildrenOfKind(String office, String projLike, String kind) {
+        switch (kind) {
+
+            case "EMBANKMENT":
+                return getEmbankmentChildren(office, projLike);
+            case "TURBINE":
+                return getTurbineChildren(office, projLike);
+            case "OUTLET":
+                return getOutletChildren(office, projLike);
+            case "LOCK":
+                return getLockChildren(office, projLike);
+            case "GATE":
+                return getGateChildren(office, projLike);
+            default:
+                return null;
+        }
+
+    }
+
+    private Map<String, List<CwmsId>> getGateChildren(String office, @Nullable String projLike) {
+        Map<String, List<CwmsId>> retval = new LinkedHashMap<>();
+        AV_GATE view = AV_GATE.AV_GATE;
```

Comment:
I think this is an un-used view, it should be av_outlet where the rating has a controlling parameter

Reply:
ok.  I switched to AV_OUTLET but I don't have data to test against for gates.

Reply:
Did you test for a controlling parameter on the rating?

Reply:
You are talking about separating the gates from the outlets - not about unit or integration tests - right?   I look for outlets that have an opening unit.   I'm not sure that is a sufficient condition but I talked to RyanM about it and there was at least one other place that also did that.